api/v1alpha1/runnerpool_types.go

```diff
@@ -0,0 +1,92 @@
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: Some lines of the code are proudly copied from Kubernetes project.
+// https://github.com/kubernetes/apimachinery/tree/master/pkg/apis/meta/v1
+// https://github.com/kubernetes/api/tree/master/core/v1
+// https://github.com/kubernetes/api/tree/master/apps/v1
+
+// RunnerPoolSpec defines the desired state of RunnerPool
+type RunnerPoolSpec struct {
+	// RepositoryName describes repository name to register Pods as self-hosted
+	// runners.
+	RepositoryName string `json:"repositoryName"`
+
+	// Number of desired pods. This is a pointer to distinguish between explicit
+	// zero and not specified. Defaults to 1.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Label selector for pods. Existing ReplicaSets whose pods are
+	// selected by this will be the ones affected by this deployment.
+	// It must match the pod template's labels.
+	Selector *metav1.LabelSelector `json:"selector"`
+
+	// Template describes the pods that will be created.
+	Template PodTemplateSpec `json:"template"`
+
+	// The deployment strategy to use to replace existing pods with new ones.
+	// +optional
+	Strategy appsv1.DeploymentStrategy `json:"strategy,omitempty"`
+}
+
+// PodTemplateSpec describes the data a pod should have when created from a template.
+// This is slightly modified from corev1.PodTemplateSpec.
+type PodTemplateSpec struct {
+	// Standard object's metadata.  The name in this metadata is ignored.
+	// +optional
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of the pod.
+	// The name of the MySQL server container in this spec must be `mysqld`.
+	Spec corev1.PodSpec `json:"spec"`
+}
+
+// ObjectMeta is metadata of objects.
+// This is partially copied from metav1.ObjectMeta.
+type ObjectMeta struct {
+	// Labels is a map of string keys and values.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Annotations is a map of string keys and values.
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// RunnerPoolStatus defines status of RunnerPool
+type RunnerPoolStatus struct {
```

Comment:
Is it possible to include the number of available pods in the status?

Reply:
In the design, the statuses of runners, which are not implemented in this task, come from `/metrics`.
I'm not sure if collecting the statuses from `/metrics` and setting it on CR is appropriate in the context of k8s programming, but it can be done also by the `RunnerPool` controller collecting the information via GitHub API,

Anyway, this is out of scope of this task, so I will consider it later.